lib/util/http.js

```diff
@@ -24,7 +24,10 @@ const isTrue = (x) => (!isBlank(x) && typeof x === 'string' && (x.toLowerCase()
 const isFalse = (x) => (!isBlank(x) && typeof x === 'string' && (x.toLowerCase() === 'false'));
 
 // Returns just the pathname of the URL, omitting querystring and other non-path decoration.
-const urlPathname = (x) => parse(x).pathname;
+const urlPathname = (url) => {
+  if (typeof url !== 'string') throw new Error(`The "url" argument must be of type string.`);
```

Comment:
I don't think this check is strictly necessary. Places where `urlPathname` is consumed, passed argument is string. But no harm in keep this.

Reply:
If `url` is `undefined` or `null`, `new URL()` will convert those to strings.  So this check avoids some potentially weird behaviour which `url.parse()` also protects against.

Reply:
```js
new URL(null, 'http://example.test').pathname
'/null'
```